src/rpdk/core/contract/resource_client.py

```diff
@@ -334,6 +334,29 @@ def generate_invalid_update_example(self, create_model):
         example = override_properties(self.invalid_strategy.example(), overrides)
         return {**create_model, **example}
 
+    def compare(self, inputs, outputs):
+        assertion_error_message = (
+            "All properties specified in the request MUST "
+            "be present in the model returned, and they MUST"
+            " match exactly, with the exception of properties"
+            " defined as writeOnlyProperties in the resource schema"
+        )
+        try:
+            for key in inputs:
+                if isinstance(inputs[key], dict):
+                    self.compare(inputs[key], outputs[key])
+                elif isinstance(inputs[key], list):
+                    assert len(inputs[key]) == len(outputs[key])
```

Comment:
is that gonna be true for all cases?

```
input = {
  "key_1": [ {"a": 10}, {"b": 11} ]
},
output = {
  "key_1": [ {"a": 10}, {"b": 11}, {"c": 12} ]
}
```
this will fail assertion but `c` is absent from input - based on cr description this should pass assertion

Reply:
but then it will fail before the actual assertion..